### PR TITLE
CI: Use Qt6 by default when available

### PR DESCRIPTION
**Pull request by RytoEX**
### Description

Use Qt6 by default when available. This applies to the Windows, macOS, and Ubuntu CI jobs.

### Motivation and Context

As announced on May 14, [we plan to use Qt6 for OBS 28](https://github.com/obsproject/obs-studio/discussions/6481) on platforms where it is available. CI and build scripts should default to Qt6 to encourage adoption of Qt6 by plugin maintainers. This will also help us during our beta and release candidate period.

As stated in the Discussion post, moving to Qt6 also allows us to stop maintaining backported patches, which reduces our maintenance burden.

### How Has This Been Tested?

I've verified that CI runs on my fork build with Qt6 when available (Windows, macOS, Ubuntu 22.04) and Qt5 when not available (Ubuntu 20.04). Flatpak will be covered in #6847.
https://github.com/RytoEX/obs-studio/actions/runs/2768057341

### Types of changes

- Bug fix (non-breaking change which fixes an issue)

- Breaking change (fix or feature that would cause existing functionality to change)

### Checklist:

- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -6,6 +6,7 @@ on:
     branches:
       - master
       - 'release/**'
+      - qt6-build
```
Comment (PatTheMav) [inline .github/workflows/main.yml:9]: Was this just for testing or do we intend to keep an official qt6-build on the main repo for the transition period?

File: CI/linux/01_install_dependencies.sh [hunk 2]
```diff
@@ -32,14 +32,25 @@ install_obs-deps() {
...
 
     sudo apt-get install -y $@
 }
 
+install_qt6-deps() {
+    shift
+    status "Install Qt6 dependencies"
+    trap "caught_error 'install_qt6-deps'" ERR
+
+    _QT6_AVAILABLE="$(sudo apt-cache search $1)"
```
Comment (PatTheMav) [inline CI/linux/01_install_dependencies.sh:48]: I think `apt-cache madison ${1}` would be the safer variant. At least last time I checked for this use case it was the suggested method.

File: CI/macos/01_install_dependencies.sh [hunk 3]
```diff
@@ -90,7 +90,7 @@ install_cef() {
     status "Set up dependency CEF v${1}"
     ensure_dir "${DEPS_BUILD_DIR}"
     unset _SKIP
-    
+
```
Comment (PatTheMav) [inline CI/macos/01_install_dependencies.sh:93]: Is that a stray tab or a removed tab? GitHub's diff view doesn't make it clear what it did here..

Comment (RytoEX) [reply CI/macos/01_install_dependencies.sh:93]: Probably removed by .editorconfig.

File: .github/workflows/main.yml (hunk 1 above)
Comment (RytoEX) [reply .github/workflows/main.yml:9]: A tired mistake, will fix.

File: CI/linux/01_install_dependencies.sh (hunk 2 above)
Comment (RytoEX) [reply CI/linux/01_install_dependencies.sh]: Changed.

File: CI/macos/01_install_dependencies.sh (hunk 3 above)
Comment (RytoEX) [reply CI/macos/01_install_dependencies.sh:93]: Discussed off-thread, and we'll leave this change in.

**Comment by SuslikV** [comment]
CI builds for Windows platform from this time will run only on Win8.1 and up (dependencies, that were updated here, relay on functions of dxgi library that are specific to newer OSes).

**Comment by RytoEX** [comment]
> CI builds for Windows platform from this time will run only on Win8.1 and up (dependencies, that were updated here, relay on functions of dxgi library that are specific to newer OSes).

That's expected and was announced in the May 14 Discussion Post, [Plan to Upgrade OBS Studio to Qt 6](https://github.com/obsproject/obs-studio/discussions/6481).

**Pull request merged**